listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -208,94 +246,158 @@ def insert(self, listens):
 
+            If neither from_ts nor to_ts is provided, the latest listens for the user are returned.
+            Returns a tuple of (listens, min_user_timestamp, max_user_timestamp)
+
             from_ts: seconds since epoch, in float
             to_ts: seconds since epoch, in float
             limit: the maximum number of items to return
             order: 0 for ASCending order, 1 for DESCending order
-            time_range: the time range (in units of 5 days) to search for listens. If none is given
-                        3 ranges (15 days) are searched. If -1 is given then all listens are searched
-                        which is slow and should be avoided if at all possible.
         """
 
-        return self.fetch_listens_for_multiple_users_from_storage([user_name], from_ts, to_ts, limit, order, time_range)
+        return self.fetch_listens_for_multiple_users_from_storage([user_name], from_ts, to_ts, limit, order)
 
-    def fetch_listens_for_multiple_users_from_storage(self, user_names: List[str], from_ts: float, to_ts: float, limit: int, order: int, time_range: int=3):
+    def fetch_listens_for_multiple_users_from_storage(self, user_names: List[str], from_ts: float, to_ts: float, limit: int, order: int):
         """ The timestamps are stored as UTC in the postgres datebase while on retrieving
             the value they are converted to the local server's timezone. So to compare
             datetime object we need to create a object in the same timezone as the server.
 
+            If neither from_ts nor to_ts is provided, the latest listens for the user are returned.
+            Returns a tuple of (listens, min_user_timestamp, max_user_timestamp)
+
             from_ts: seconds since epoch, in float
             to_ts: seconds since epoch, in float
             limit: the maximum number of items to return
             order: 0 for DESCending order, 1 for ASCending order
-            time_range: the time range (in units of 5 days) to search for listens. If none is given
-                        3 ranges (15 days) are searched. If -1 is given then all listens are searched
-                        which is slow and should be avoided if at all possible.
         """
 
-        if time_range is None:
-            time_range = 3
-
-        if time_range < 0:
-            max_timestamp_window = -1
-        else:
-            max_timestamp_window = SECONDS_IN_TIME_RANGE * time_range
-            if to_ts is None:
-                to_ts = from_ts + max_timestamp_window
-            elif from_ts is None:
-                from_ts = to_ts - max_timestamp_window
-
+        min_user_ts = max_user_ts = -1
+        for user_name in user_names:
+            min_ts, max_ts = self.get_timestamps_for_user(user_name)
+            if min_user_ts < 0:
+                min_user_ts = min_ts
+                max_user_ts = max_ts
+            if min_ts and min_ts < min_user_ts:
```

Comment:
Could this loop/check be improved by doing something like

    min_user_ts = min(min_ts, min_user_ts)
    max_user_ts = max(max_ts, max_user_ts)

I would set the sentinels to None before the loop (to make it more explicit that this is an empty value), and have an additional check like you do on line 306.

Reply:
Reworked to be more concise.